backend/src/main/kotlin/com/tngtech/apicenter/backend/connector/rest/controller/ServiceController.kt

```diff
@@ -52,6 +54,28 @@ class ServiceController @Autowired constructor(
         return specificationFileDtoMapper.fromDomain(specification)
     }
 
+    @PutMapping("/{serviceId}/chmod/{userId}")
+    fun chmodService(@PathVariable serviceId: String,
+                     @PathVariable userId: String,
+                     @RequestParam(value = "view", defaultValue = "false") view: String,
+                     @RequestParam(value = "viewPrereleases", defaultValue = "false") viewPrereleases: String,
+                     @RequestParam(value = "edit", defaultValue = "false") edit: String
+    ) {
+        val id = ServiceId(serviceId)
+        serviceHandler.changePermission(id, userId, view.toBoolean(), PermissionType.VIEW)
+        serviceHandler.changePermission(id, userId, viewPrereleases.toBoolean(), PermissionType.VIEWPRERELEASE)
+        serviceHandler.changePermission(id, userId, edit.toBoolean(), PermissionType.EDIT)
+    }
+
+    @GetMapping("/{serviceId}/getmod/{userId}")
+    fun getmodService(@PathVariable serviceId: String,
```

Comment:
Why do we need this? The controller should already check whether the current user has the necessary permissions before it returns the spec.

Reply:
This is only used in the edit-permissions frontend form, to pre-populate the permissions of a certain user. It is not used for authorization anywhere (see the demo gif)